assets/app/scripts/services/data.js

```diff
@@ -430,6 +439,83 @@ angular.module('openshiftConsole')
     };
   };
 
+// resource:  API resource (e.g. "pods")
+// name:      API name, the unique name for the object
+// context:   API context (e.g. {project: "..."})
+// callback:  optional function to be called with the initial list of the requested resource,
+//            and when updates are received, parameters passed to the callback:
+//            obj:    the requested object
+//            event:  specific event that caused this call ("ADDED", "MODIFIED",
+//                    "DELETED", or null) callbacks can optionally use this to
+//                    more efficiently process updates
+// opts:      options
+//            poll:   true | false - whether to poll the server instead of opening
+//                    a websocket. Default is false.
+//            pollInterval: in milliseconds, how long to wait between polling the server
+//                    only applies if poll=true.  Default is 5000.
+//
+// returns handle to the watch, needed to unwatch e.g.
+//        var handle = DataService.watch(resource,context,callback[,opts])
+//        DataService.unwatch(handle)
+  DataService.prototype.watchObject = function(resource, name, context, callback, opts) {
+    resource = normalizeResource(resource);
+    opts = opts || {};
+
+    var wrapperCallback;
+    if (callback) {
+      // If we were given a callback, add it
+      this._watchObjectCallbacks(resource, name, context).add(callback);
+      var self = this;
+      wrapperCallback = function(items, event, item) {
+        // If we got an event for a single item, only fire the callback if its the item we care about
+        if (item && item.metadata.name === name) {
+          self._watchObjectCallbacks(resource, name, context).fire(item, event);
+        }
+        else {
+          // Otherwise see if we can find the item we care about in the list
+          angular.forEach(items.by("metadata.name"), function(obj){
+            if (obj && obj.metadata.name === name) {
+              self._watchObjectCallbacks(resource, name, context).fire(obj, event);
```

Comment:
Actually dont even need a loop here it should basically be `if (items.by("metadata.name")[name])`